### ISSUE TITLE
Misleading error message for 403/Forbidden response from API

### ISSUE THREAD

**Issue by GinkgoFJG** [cividesk/com.cividesk.email.sparkpost#5]
Related to #4, I think I'd have gotten to the bottom of my configuration problem faster with more nuanced handling of the response. For 403/Forbidden, the extension reports:

> Check that the API key is valid.

... which suggests to me that the API key doesn't exist on the server (i.e., analogous to a user name/password mismatch). What it actually means, though, is that your valid API key doesn't have permission to do the requested action.

If I'm not mistaken -- and I could be, because I didn't find good documentation around the return codes for this API -- 401/Unauthorized means the API key is bogus. Did you find better documentation for return codes than [this](https://support.sparkpost.com/customer/en/portal/articles/2140916-extended-error-codes)?

Do you agree with my assessment? If so, I'll gladly make a PR to add some clarity to the feedback messages.

**Comment by nganivet** [comment]
Frank - It seems like I am already differentiating between invalid and unauthoirized API key at:https://github.com/cividesk/com.cividesk.email.sparkpost/blob/master/CRM/Sparkpost.php#L98. So you should have had the correct message.

I would like to get this corrected since not granting the right privileges to the API key is a common mistake. Can you be more specific about what you did, what got displayed on screen, and what should we do instead for a seamless user experience?

**Comment by GinkgoFJG** [comment]
Nicolas, thanks for updating the documentation. I think when I created the account it popped up a message to the effect of, "We're sure you want to get started -- copy this API key to get started sending mail." Since the process was so transparent, I didn't get a chance to configure the permissions for this key until I stumbled upon it in the SparkPost UI. Perhaps it should have been obvious that I needed to adjust the permissions, but it didn't occur to me.

The message I got was "Check that the API key is valid." Perhaps a better message would be, "Check that the API key is valid and that it has been assigned permission to perform this action"?

Anyway, I think the documentation update might be enough to set people on the right path.
